frontend/catalyst/utils/jax_extras.py

```diff
@@ -636,3 +636,16 @@ def _gather_shape_rule_dynamic(
             )
 
     return _gather_shape_computation(indices, dimension_numbers, slice_sizes)
+
+
+def remove_host_context(jaxpr):
+    """This function will remove the host context.
+
+    It is expected that callers to this function do not use the host context
+    and instead forward the arguments directly to the qnode.
+    This is useful for example in the proof-of-concept for cuda integration.
+
+    Later iterations **might** have a host context. This is something not currently planned.
+    """
+    # Is this always the case, let's try with different parameters.
+    return jaxpr.jaxpr.eqns[0][3]["call_jaxpr"]
```

Comment:
This seems hacky :D is there any alternative to this nested indexing to access the `call_jaxpr`?

Reply:
I think we can leave this as a TODO for another PR? We can do an analysis that checks that there is a single call and all parameters are forwarded and all return variables are returned. If these conditions are true, then the above equation will always be correct.